### PR TITLE
Updated Tagging UI

### PR DESCRIPTION
**Pull request by jkppr**
**IMPORTANT:** ~~This PR is blocked by the missing backend API endpoint for removing tags!~~ See #2689 #2605 
=> Endpoint added in #2724 

This PR updates the UX of the event tagging feature and adds the option to remove tags from events.
* Tags are now sorted. Quick tags first, rest alphabetically.
* Tags can now be dynamically added and removed without the need of confirming the dialog.
* Applied tags can be removed from the EventDetail view or via the EventTagMenu.
* The list for new tags will only show available tags.
* The tag menu also supports timelines with events that have no tag attribute now.

Creating this PR as a draft for review and feedback until the backend API is ready.

Demo: 
TS_tag_demo

### CODE REVIEW

**Comment by jkppr** [comment]
The matching API endpoint is added in #2724 . 
I updated the PR to work with the endpoint appropriately.

File: timesketch/frontend-ng/src/components/Explore/EventTagMenu.vue [hunk 1]
```diff
@@ -80,30 +116,65 @@ export default {
...
-        this.showMenu = false
+      }
+
+      ApiClient.tagEvents(this.sketch.id, [this.event], [tagToAdd])
+        .then((response) => {
+          this.$set(this.event._source.tag, this.event._source.tag.length, tagToAdd)
+          this.$store.dispatch('updateTimelineTags', { sketchId: this.sketch.id, tag: tagToAdd, num: 1 })
+        })
+        .catch((e) => {
+          console.error('Cannot tag event')
```
Comment (berggren) [inline timesketch/frontend-ng/src/components/Explore/EventTagMenu.vue:173]: Maybe add the error here as well?

File: timesketch/frontend-ng/src/components/Explore/EventTags.vue [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<template>
+  <span>
+    <v-chip
+      small
+      class="mr-1"
+      v-for="tag in sortedTags"
+      :close="showDetails ? true : false"
```
Comment (berggren) [inline timesketch/frontend-ng/src/components/Explore/EventTags.vue:19]: could we wrap this in a `<hover>` instead?

**Review by berggren — CHANGES_REQUESTED** [review]

File: timesketch/frontend-ng/src/components/Explore/EventTagMenu.vue [hunk 3]
```diff
@@ -30,27 +33,59 @@ limitations under the License.
             :text-color="tag.textColor"
             class="text-center"
             small
+            :close="showClose(tag.tag)"
             @click="addTags(tag.tag)"
+            @click:close="removeTags(tag.tag)"
```
[the remark this answers is not in the document]
Comment (jkppr) [reply timesketch/frontend-ng/src/components/Explore/EventTagMenu.vue:38]: good point

File: timesketch/frontend-ng/src/components/Explore/EventTagMenu.vue [hunk 4]
```diff
@@ -80,30 +116,65 @@ export default {
...
+      customTags = customTags.filter((tag) => !this.customSelectedTags.includes(tag))
+      customTags.sort((a, b) => { return a.localeCompare(b)})
+      return customTags
+    },
   },
   methods: {
-    addTags: function (tagsToAdd) {
-      if (!Array.isArray(tagsToAdd)) {
-        tagsToAdd = [tagsToAdd]
+    showClose(tag) {
```
[the remark this answers is not in the document]
Comment (jkppr) [reply timesketch/frontend-ng/src/components/Explore/EventTagMenu.vue:138]: sure, works as well.

File: timesketch/frontend-ng/src/components/Explore/EventTags.vue (hunk 2 above)
Comment (jkppr) [reply timesketch/frontend-ng/src/components/Explore/EventTags.vue:19]: Yeah, that works as well. However, with the hover option the table jumps a bit around when hovering. Please have a look and let me know what you think.

File: timesketch/frontend-ng/src/components/Explore/EventTags.vue [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+<script>
+import ApiClient from '../../utils/RestApiClient'
+export default {
+  props: ['item', 'tagConfig', 'showDetails'],
+  computed: {
+    sketch() {
+      return this.$store.state.sketch
+    },
+    sortedTags() {
```
[the remark this answers is not in the document]
Comment (jkppr) [reply timesketch/frontend-ng/src/components/Explore/EventTags.vue:44]: Ack, will be reworked when we make quickTags configurable. I added a comment.

[the remark this answers is not in the document]
Comment (jkppr) [reply timesketch/frontend-ng/src/components/Explore/EventTags.vue:44]: Ack, will be reworked when we make quickTags configurable. I added a comment.

File: timesketch/frontend-ng/src/components/Explore/EventTags.vue (hunk 2 above)
Comment (jkppr) [reply timesketch/frontend-ng/src/components/Explore/EventTags.vue]: fixed by adding a 32px padding on non-hover.

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**